src/sinks/datadog_metrics.rs

```diff
@@ -0,0 +1,377 @@
+            builder.uri(uri.clone());
+
+            builder.header("Content-Type", "application/json");
+            builder.body(body).unwrap()
+        })?;
+
+        let datadog_http_service = DatadogSvc {
+            config,
+            state: DatadogState {
+                last_sent_timestamp: Utc::now().timestamp(),
+            },
+            inner: http_service,
+        };
+
+        let service = ServiceBuilder::new()
+            .concurrency_limit(in_flight_limit)
+            .rate_limit(rate_limit_num, Duration::from_secs(rate_limit_duration))
+            .retry(policy)
+            .timeout(Duration::from_secs(timeout))
+            .service(datadog_http_service);
+
+        let sink = BatchServiceSink::new(service, acker).batched_with_min(
+            Vec::new(),
+            batch_size,
+            Duration::from_secs(batch_timeout),
+        );
+
+        Ok(Box::new(sink))
+    }
+
+    fn healthcheck(config: DatadogConfig) -> crate::Result<super::Healthcheck> {
+        let uri = format!("{}/api/v1/validate?api_key={}", config.host, config.api_key)
+            .parse::<Uri>()
+            .context(super::UriParseError)?;
+
+        let request = hyper::Request::get(uri).body(hyper::Body::empty()).unwrap();
+
+        let https = HttpsConnector::new(4).expect("TLS initialization failed");
+        let client = hyper::Client::builder().build(https);
+
+        let healthcheck = client
+            .request(request)
+            .map_err(|err| err.into())
+            .and_then(|response| match response.status() {
+                StatusCode::OK => Ok(()),
+                other => Err(super::HealthcheckError::UnexpectedStatus { status: other }.into()),
+            });
+
+        Ok(Box::new(healthcheck))
+    }
+}
+
+impl Service<Vec<Event>> for DatadogSvc {
+    type Response = HttpResponse;
+    type Error = HttpError;
+    type Future = Box<dyn Future<Item = Self::Response, Error = Self::Error> + Send + 'static>;
+
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        self.inner.poll_ready()
+    }
+
+    fn call(&mut self, items: Vec<Event>) -> Self::Future {
+        let now = Utc::now().timestamp();
+        let interval = now - self.state.last_sent_timestamp;
+        self.state.last_sent_timestamp = now;
+
+        let input = encode_events(items, interval, &self.config.namespace);
+        let body = serde_json::to_vec(&input).unwrap();
+
+        self.inner.call(body)
+    }
+}
+
+fn encode_tags(tags: HashMap<String, String>) -> Vec<String> {
+    let mut pairs: Vec<_> = tags
+        .iter()
+        .map(|(name, value)| format!("{}:{}", name, value))
+        .collect();
+    pairs.sort();
+    pairs
+}
+
+fn encode_timestamp(timestamp: Option<DateTime<Utc>>) -> i64 {
+    if let Some(ts) = timestamp {
+        ts.timestamp()
+    } else {
+        Utc::now().timestamp()
+    }
+}
+
+fn encode_namespace(namespace: &str, name: String) -> String {
+    if !namespace.is_empty() {
```

Comment:
If we allow this to be empty, should we make it optional in the config?

Reply:
I'd like to encourage users to fill the namespace config this way, because it's a really useful practice.

But in case user has reasons to not use namespace (maybe working with some legacy pipeline), I'd like to make it explicit in the config.